output/fluentdforward/config.go

```diff
@@ -68,22 +68,23 @@ func (cfg *Config) NewSerializer(parentLogger logger.Logger, schema base.LogSche
 // NewChunkMaker creates LogChunkMaker
 func (cfg *Config) NewChunkMaker(parentLogger logger.Logger, tag string) base.LogChunkMaker {
 	var asArray bool
-	var initCompessor shared.InitCompessorFunc
 
+	var initCompressorFunc shared.InitCompessorFunc
 	switch cfg.MessageMode {
 	case forwardprotocol.ModeForward:
 		asArray = true
 	case forwardprotocol.ModePackedForward:
 	case forwardprotocol.ModeCompressedPackedForward:
-		initCompessor = shared.InitGzipCompessor
+		initCompressorFunc = shared.InitGzipCompessor
 	default:
 		parentLogger.Fatalf("unsupported message mode: %s", cfg.MessageMode)
 	}
 
 	encoder := newEncoder(tag, asArray, msgBufCapacity)
-	basicChunkFactory := shared.NewChunkFactory(parentLogger, chunkIDSuffix, msgBufCapacity, initCompessor, encoder)
+	newChunkFunc := buildNewChunkFunc(parentLogger, initCompressorFunc, encoder, chunkMaxRecords, chunkMaxSizeBytes)
```

Comment:
Would it be cleaner if `chunkMaxRecords` and `chunkMaxSizeBytes` are moved to `chunk.go` and become constants, instead of per-chunk variables? It was perhaps passed here previously when parameters are centralized in def/params, but no longer so.

Also the same with Datadog's

Reply:
Yea, initially they were baked into `chunk.go`, but using production values for unit testing isn't the best idea, especially with `chunkMaxSizeBytes`, which is very large (5MB), so if we want to test this max size flush behavior - it becomes very inconvenient to have consts, which is why I pass them to `buildNewChunkFunc`, so that they can be overridden in tests if needed.